source/WHIMChannel.mc

```diff
@@ -73,9 +73,25 @@ class WHIMChannel extends Ant.GenericChannel
             var payload = msg.getPayload();
             
             if (Ant.MSG_ID_BROADCAST_DATA == msg.messageId) {
-                System.println( "Broadcast data received" );
-                // TODO: Handle broadcast data messages
-
+                
+                // Handle Data Page 1
+                if (payload[0] == 0x01) {
+                	System.println("Data Page 1 Received!");
+                	
+	                if (current_view_id == PAIR)
+	                {
+	                	System.println("View switched!");
+				        var view = new SensorDetailsView();
+				        WatchUi.switchToView(view, new SensorDetailsDelegate(view), WatchUi.SLIDE_IMMEDIATE);
+	                }
+	                
+	                if (impacts != payload[1]) {
+	                	System.println("New Data!");
+			        	impacts = payload[1];
+			        	WatchUi.requestUpdate();
+			        }
+			    }
```

Comment:
Add an else case, which prints that an undefined/unrecognized data page has been received

Reply:
Done